### DISCUSSION TITLE
Sensor is not able to connect with Manager

### DISCUSSION THREAD

**Discussion by arjoshi89** [Security-Onion-Solutions/securityonion#1642] [Unsupported Versions] [answered]
Deploying New Distributed Nodes.
Issue: Sensor is not able to connect with Manager. Issue happens with version 2.3.1 ISO and 2.3 ISO both

Found below error under sosetup.log of sensor:
"Attempt to authenticate with the salt master failed with timeout error"

Checked on vmware workstation and physical HP Hardware both. Sensor and Manager can ping each other.
Uninstall and install setup many times. Reboot many times.

If I use 16.04 ISO then there is no issue with same setup.

**Comment by TOoSmOotH** [comment]
It should try and SSH to it first. Are you prompted for the soremote credentials?

**Comment by arjoshi89** [comment, marked as answer]
Thank you for replying. 

Yes, I received prompt on sensor for soremote credentials. I confirm that I put correct credentials. 

As per tcpdump on Manager, I can see traffic is coming in and going out for port 22 when sensor try to connect. No Issue.

192.168.10.3 - Manager IP
192.168.10.4 - Sensor IP

Thereafter, Sensor sends Syn request on port 4506 to Manager but Manager is responding as below:
192.168.10.3 > 192.168.10.4: ICMP host 192.168.10.3 unreachable - admin prohibited, length 68

At same time, Sensor screen shows process of "Accepting salt key on manager"

**Reply by TOoSmOotH** [reply]
There are a couple of things that happen. The first thing is once it connects via SSH it should be adding the host to the firewall rules. Do you see the host in /opt/so/saltstack/local/salt/firewall/hostgroups.local.yaml under the minions group?

**Reply by arjoshi89** [reply]
Thank you, I was able to figure it out. Now, Sensor is connecting. Out of curiosity, What is difference between sensor, manager and search node? We will be deploying it at higher traffic place (1 search, 1 manager, 1 sensor). For which node, we have to keep higher storage for logs? Which node keeps all copy of logs and capture?
I referred online documentation but it was still confusing.

**Reply by dougburks** [reply]
Forward nodes run sensor processes and forward most sensor logs to the manager (although full packet capture remains on the forward node). Search nodes then pull logs from the queue on the manager and parse and index them.

**Reply by arjoshi89** [reply]
Thank you !!

**Reply by 4revern00b** [reply]
@arjoshi89 hey, can you share how you managed to resolve the issue? I am having a similar issue.

**Reply by arjoshi89** [reply]
I found port 4506 conflict during my setup and I only got success with 2.3.1 iso.
